pkg/apb/dockerhub_registry.go

```diff
@@ -173,20 +169,78 @@ func (r *DockerHubRegistry) loadBundleImageData(org string) ([]*ImageData, error
 	}
 
 	channel := make(chan *ImageData)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	// Will kick of go routines for each result in the images.
+	loadImagesFromResult := func(ctx context.Context, images []*Images) {
+		for _, imageName := range images {
+			r.log.Debugf("Trying to load %v/%v", imageName.Namespace, imageName.Name)
+			go r.loadImageData(ctx, "docker://"+imageName.Namespace+"/"+imageName.Name, channel)
+		}
+	}
+
+	// getNextImages - will follow the next URL using go routines.
+	// The workflow is query the url, if Next is defined then kickoff another go routine to get those images.
+	// then kick of the loading of image data.
+	var getNextImages func(ctx context.Context, org, token, url string, ch chan<- *ImageData, cancelFunc context.CancelFunc)
+
+	getNextImages = func(ctx context.Context, org, token, url string, ch chan<- *ImageData, cancelFunc context.CancelFunc) {
+		req, err = http.NewRequest("GET", url, nil)
+		if err != nil {
+			r.log.Errorf("unable to get next images for url: %v - %v", url, err)
+			cancelFunc()
+			close(ch)
+			return
+		}
+		req.Header.Set("Authorization", fmt.Sprintf("JWT %v", token))
```

Comment:
This code does the same thing as lines 138-163. Maybe we can consolidate that piece into another function that each can use.